broker/src/main/java/io/camunda/zeebe/broker/transport/streamapi/JobStreamApiServer.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.transport.streamapi;
+
+import io.atomix.cluster.ClusterMembershipEvent;
+import io.atomix.cluster.ClusterMembershipEvent.Type;
+import io.atomix.cluster.ClusterMembershipEventListener;
+import io.atomix.cluster.MemberId;
+import io.atomix.cluster.messaging.ClusterCommunicationService;
+import io.camunda.zeebe.protocol.impl.stream.AddStreamRequest;
+import io.camunda.zeebe.protocol.impl.stream.JobStreamTopics;
+import io.camunda.zeebe.protocol.impl.stream.RemoveStreamRequest;
+import io.camunda.zeebe.scheduler.Actor;
+import io.camunda.zeebe.stream.api.JobActivationProperties;
+import io.camunda.zeebe.util.buffer.BufferReader;
+import java.util.function.Function;
+import org.agrona.concurrent.UnsafeBuffer;
+
+public final class JobStreamApiServer extends Actor implements ClusterMembershipEventListener {
+  private final ClusterCommunicationService transport;
+  private final StreamApiHandler<JobActivationProperties> requestHandler;
+
+  public JobStreamApiServer(
+      final ClusterCommunicationService transport,
+      final StreamApiHandler<JobActivationProperties> requestHandler) {
+    this.transport = transport;
+    this.requestHandler = requestHandler;
+  }
+
+  @Override
+  public boolean isRelevant(final ClusterMembershipEvent event) {
```

Comment:
:+1: Didn't know this exists :D

Reply:
Me neither, so I can't fully guarantee it works :smile: